app/Helpers/Payment/Donation.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Helpers\Payment;
+
+final class Donation implements PaymentInterface
+{
+    private const DESCRIPTION = 'Donatie aan Anderwijs. Bedankt%s!';
+
+    private float $totalAmount;
+
+    private ?string $name;
+
+    public function __construct(float $totalAmount, ?string $name)
+    {
+        $this->totalAmount = $totalAmount;
+        $this->name = $name;
+    }
+
+    public function getTotalAmount(): float
+    {
+        return $this->totalAmount;
+    }
+
+    public function getDescription(): string
+    {
+        $name = $this->name !== null ? ' ' . $this->name : '';
+        return sprintf(self::DESCRIPTION, $name);
+    }
+
+    public function getCurrency(): string
+    {
+        return 'EUR';
+    }
+
+    public function getMetadata()
+    {
+        return [
+            'payment_type' => 'donation',
+            'name' => $this->name,
+        ];
+    }
+
+    public function getRedirectUrl(): string
+    {
+        return action('iDealController@genericResponse');
```

Comment:
Volgens mij verwijst deze nu in de controller door naar de **registration** responspagina, dat is een beetje vreemd als je alleen doneert toch? En als we de responspagina in `pages` willen gebruiken, moeten we die nog wat verder oppoetsen denk ik.